python/rmm/_lib/device_buffer.pyx

```diff
@@ -50,6 +55,20 @@ cdef class DeviceBuffer:
         buf.c_obj = move(ptr)
         return buf
 
+    cpdef bytes tobytes(self, uintptr_t stream=0):
+        cdef const device_buffer* dbp = self.c_obj.get()
+        cdef bytes b = PyBytes_FromStringAndSize(NULL, self.c_size())
+        cdef char* p = PyBytes_AS_STRING(b)
+        cdef cudaError_t err
+
+        with nogil:
+            copy_to_host(dbp[0], <void*>p, <cudaStream_t>stream)
+            err = cudaStreamSynchronize(<cudaStream_t>stream)
+        if err != cudaSuccess:
+            raise RuntimeError("Stream sync failed.")
```

Comment:
It would be good to give the actual error code here if possible.

Reply:
Good idea. Added in the latest commit 🙂